Subject: [PATCH] Additions: Linux: vboxvideo: Additional build fixes for RHEL
 9.1 and 9.2 kernels, bugref:10190, bugref:10237, bugref:10279,
 ticketref:21446, ticketref:21450.

Added centos (8.8) stream

diff --git a/trunk/src/VBox/Additions/linux/drm/vbox_drv.c b/trunk/src/VBox/Additions/linux/drm/vbox_drv.c
index b3f22e56e39..b85f4b9a618 100644
--- a/vboxvideo/vbox_drv.c
+++ b/vboxvideo/vbox_drv.c
@@ -52,7 +52,7 @@
 
 /** Detect whether kernel mode setting is OFF. */
 #if defined(CONFIG_VGA_CONSOLE)
-# if RTLNX_VER_MIN(5,17,0)
+# if RTLNX_VER_MIN(5,17,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MIN(9,1)
 #  define VBOX_VIDEO_NOMODESET() drm_firmware_drivers_only() && vbox_modeset == -1
 # elif RTLNX_VER_MIN(4,7,0)
 #  define VBOX_VIDEO_NOMODESET() vgacon_text_force() && vbox_modeset == -1
diff --git a/trunk/src/VBox/Additions/linux/drm/vbox_drv.h b/trunk/src/VBox/Additions/linux/drm/vbox_drv.h
index 0a5a5f051d8..2902865848d 100644
--- a/vboxvideo/vbox_drv.h
+++ b/vboxvideo/vbox_drv.h
@@ -185,7 +185,7 @@
 # include <drm/ttm/ttm_resource.h>
 #endif
 
-#if RTLNX_VER_MIN(6,0,0)
+#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
 # include <drm/drm_framebuffer.h>
 #endif
 
diff --git a/trunk/src/VBox/Additions/linux/drm/vbox_mode.c b/trunk/src/VBox/Additions/linux/drm/vbox_mode.c
index c6b7ef3143e..9f4316dd55b 100644
--- a/vboxvideo/vbox_mode.c
+++ b/vboxvideo/vbox_mode.c
@@ -46,7 +46,7 @@
 # include <drm/drm_probe_helper.h>
 #endif
 
-#if RTLNX_VER_MIN(6,0,0)
+#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
 # include <drm/drm_edid.h>
 #endif
 
diff --git a/trunk/src/VBox/Additions/linux/drm/vbox_ttm.c b/trunk/src/VBox/Additions/linux/drm/vbox_ttm.c
index 5cc20cb8edd..abb027fcb88 100644
--- a/vboxvideo/vbox_ttm.c
+++ b/vboxvideo/vbox_ttm.c
@@ -316,15 +316,16 @@ static struct ttm_tt *vbox_ttm_tt_create(struct ttm_buffer_object *bo,
 #if RTLNX_VER_MAX(5,10,0) && !RTLNX_RHEL_RANGE(8,5, 8,99)
 	tt->func = &vbox_tt_backend_func;
 #endif
-#if RTLNX_VER_MAX(4,17,0) && !RTLNX_RHEL_MAJ_PREREQ(7,6) && !RTLNX_SUSE_MAJ_PREREQ(15,1) && !RTLNX_SUSE_MAJ_PREREQ(12,5)
-	if (ttm_tt_init(tt, bdev, size, page_flags, dummy_read_page)) {
-#elif RTLNX_VER_MAX(5,11,0) && !RTLNX_RHEL_RANGE(8,5, 8,99)
-	if (ttm_tt_init(tt, bo, page_flags)) {
-#elif RTLNX_VER_MAX(5,19,0)
+#if RTLNX_VER_MIN(5,19,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
+	if (ttm_tt_init(tt, bo, page_flags, ttm_write_combined, 0)) {
+#elif RTLNX_VER_MIN(5,11,0) || RTLNX_RHEL_RANGE(8,5, 8,8)
 	if (ttm_tt_init(tt, bo, page_flags, ttm_write_combined)) {
+#elif RTLNX_VER_MIN(4,17,0) || RTLNX_RHEL_MAJ_PREREQ(7,6) || RTLNX_SUSE_MAJ_PREREQ(15,1) || RTLNX_SUSE_MAJ_PREREQ(12,5)
+	if (ttm_tt_init(tt, bo, page_flags)) {
 #else
-	if (ttm_tt_init(tt, bo, page_flags, ttm_write_combined, 0)) {
+	if (ttm_tt_init(tt, bdev, size, page_flags, dummy_read_page)) {
 #endif
+
 		kfree(tt);
 		return NULL;
 	}
-- 
2.39.2

